src/PersonObjects/Person.ts

```diff
@@ -40,18 +43,195 @@ export abstract class Person implements IPerson {
+      return;
+    }
+    if (Player.sourceFileLvl(5) > 0 || this.skills.intelligence > 0 || Player.bitNodeN === 5) {
+      this.exp.intelligence += exp;
+      this.skills.intelligence = Math.floor(this.calculateSkill(this.exp.intelligence, 1));
+    }
+  }
+
+  gainStats(retValue: WorkStats): void {
+    this.gainHackingExp(retValue.hackExp * this.mults.hacking_exp);
+    this.gainStrengthExp(retValue.strExp * this.mults.strength_exp);
+    this.gainDefenseExp(retValue.defExp * this.mults.defense_exp);
+    this.gainDexterityExp(retValue.dexExp * this.mults.dexterity_exp);
+    this.gainAgilityExp(retValue.agiExp * this.mults.agility_exp);
+    this.gainCharismaExp(retValue.chaExp * this.mults.charisma_exp);
+    this.gainIntelligenceExp(retValue.intExp);
+  }
+
+  regenerateHp(amt: number): void {
+    if (typeof amt !== "number") {
+      console.warn(`Player.regenerateHp() called without a numeric argument: ${amt}`);
+      return;
+    }
+    this.hp.current += amt;
+    if (this.hp.current > this.hp.max) {
+      this.hp.current = this.hp.max;
+    }
+  }
+
+  updateSkillLevels(this: Person): void {
+    this.skills.hacking = Math.max(
+      1,
+      Math.floor(this.calculateSkill(this.exp.hacking, this.mults.hacking * currentNodeMults.HackingLevelMultiplier)),
+    );
+    this.skills.strength = Math.max(
+      1,
+      Math.floor(
+        this.calculateSkill(this.exp.strength, this.mults.strength * currentNodeMults.StrengthLevelMultiplier),
+      ),
+    );
+    this.skills.defense = Math.max(
+      1,
+      Math.floor(this.calculateSkill(this.exp.defense, this.mults.defense * currentNodeMults.DefenseLevelMultiplier)),
+    );
+    this.skills.dexterity = Math.max(
+      1,
+      Math.floor(
+        this.calculateSkill(this.exp.dexterity, this.mults.dexterity * currentNodeMults.DexterityLevelMultiplier),
+      ),
+    );
+    this.skills.agility = Math.max(
+      1,
+      Math.floor(this.calculateSkill(this.exp.agility, this.mults.agility * currentNodeMults.AgilityLevelMultiplier)),
+    );
+    this.skills.charisma = Math.max(
+      1,
+      Math.floor(
+        this.calculateSkill(this.exp.charisma, this.mults.charisma * currentNodeMults.CharismaLevelMultiplier),
+      ),
+    );
+
+    const ratio: number = Math.min(this.hp.current / this.hp.max, 1);
+    this.hp.max = Math.floor(10 + this.skills.defense / 10);
+    this.hp.current = Math.round(this.hp.max * ratio);
+  }
+
+  hasAugmentation(augName: string, ignoreQueued = false) {
+    if (this.augmentations.some((a) => a.name === augName)) {
+      return true;
+    }
+    if (!ignoreQueued && this.queuedAugmentations.some((a) => a.name === augName)) {
+      return true;
+    }
+    return false;
+  }
+
+  travel(cityName: CityName): boolean {
+    if (!Player.canAfford(CONSTANTS.TravelCost)) {
+      return false;
+    }
+
+    Player.loseMoney(CONSTANTS.TravelCost, this instanceof PlayerObject ? "other" : "sleeves");
```

Comment:
Not for this PR (let's keep this one simple), but Person.ts has a dependency on PlayerObject that should be wholly unnecessary (and is a rather nasty dependency loop) because of this line. This is extra gross because what's being done here can be trivially done by adding a new method to Person itself to get the `moneySourceLabel()`.